Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -957,11 +963,23 @@ bool MatrixWorkspace::isCommonBins() const {
   if (!m_isCommonBinsFlagSet) {
     m_isCommonBinsFlag = true;
 
+    const size_t numHist = getNumberHistograms();
+    bool oneNonEmpty = false;
+    if (numHist > 1) {
+      !readY(0).empty();
+      for (size_t i = 1; i < numHist; ++i) {
+        if (!readY(0).empty()) {
+          oneNonEmpty = true;
```

Comment:
Maybe unit unit test should be added for the case of empty histograms?

Reply:
`ValidateCommonBoundaries` already has one

Reply:
@peterfpeterson Sorry, can't find a test covering this new special case, link?

Reply:
It is CommonBinsValidatorTest L20. I'm adding a couple of tests in MatrixWorkspace

Reply:
Yes but that does not seem to cover the new code, namely cases where some of the histograms are empty while others are not?